src/internal/NeoEspRmtMethod.h

```diff
@@ -0,0 +1,293 @@
+License along with NeoPixel.  If not, see
+<http://www.gnu.org/licenses/>.
+-------------------------------------------------------------------------*/
+
+#pragma once
+
+#ifdef ARDUINO_ARCH_ESP32
+
+#ifndef ICACHE_RAM_ATTR
+#define ICACHE_RAM_ATTR IRAM_ATTR
+#endif
+
+extern "C"
+{
+#include <Arduino.h>
+#include <driver/gpio.h>
+#include <driver/periph_ctrl.h>
+#include <driver/rmt.h>
+}
+
+// -- RMT clock divider
+#define CLK_DIV_RMT     8   // pulse granularity ->  4 =(50ns) 8 =(100ns)
+
+// -- RMT cycles conversion
+#define F_CLK_RMT       (   80000000L )
+#define NS_PER_SEC      ( 1000000000L )
+#define CYCLES_PER_SEC  ( F_CLK_RMT / CLK_DIV_RMT )
+#define NS_PER_CYCLE    ( NS_PER_SEC / CYCLES_PER_SEC )
+#define NS_TO_CYCLES(n) ( (n) / NS_PER_CYCLE )
+#define CYCLES_TO_NS(n) ( (n) * NS_PER_CYCLE )
+#define CYCLES_TO_US(n) ( CYCLES_TO_NS(n) / 1000L )
+
+class NeoEspRmtSpeedWs2813
+{
+public:
+    const static uint32_t T0H = 300;
+    const static uint32_t T0L = 900;
+    const static uint32_t T1H = 900;
+    const static uint32_t T1L = 300;
+    const static uint32_t ResetTimeUs = 300;
+};
+
+class NeoEspRmtSpeedSK6812
+{
+public:
+    const static uint32_t T0H = 300;
+    const static uint32_t T0L = 900;
+    const static uint32_t T1H = 600;
+    const static uint32_t T1L = 600;
+    const static uint32_t ResetTimeUs = 80;
+};
+
+class NeoEspRmtSpeed800Kbps
+{
+public:
+    const static uint32_t T0H = 300;
+    const static uint32_t T0L = 900;
+    const static uint32_t T1H = 900;
+    const static uint32_t T1L = 300;
+    const static uint32_t ResetTimeUs = 300;
+};
+
+class NeoEspRmtSpeed400Kbps
+{
+public:
+    const static uint32_t T0H = 700;
+    const static uint32_t T0L = 1800;
+    const static uint32_t T1H = 1800;
+    const static uint32_t T1L = 700;
+    const static uint32_t ResetTimeUs = 300;
+};
+
+
+enum NeoRmtState
+{
+    NeoRmtState_Idle,
+    NeoRmtState_Pending,
+    NeoRmtState_Sending,
+};
+
+
+template<typename T_SPEED> class NeoEspRmtMethodBase
+{
+public:
+    NeoEspRmtMethodBase(uint8_t pin, uint16_t pixelCount, size_t elementSize) :
+        _pin(pin)
+    {
+        _pixelsSize = pixelCount * elementSize;
+        _bufferSize = _pixelsSize + 1; // +1 for RESET pulse
+
+        _pixels = (uint8_t*)malloc(_pixelsSize);
+        memset(_pixels, 0x00, _pixelsSize);
+
+        _rmtBuffer = (uint8_t*)malloc(_bufferSize);
+        memset(_rmtBuffer, 0x00, _bufferSize);
+    }
+
+    ~NeoEspRmtMethodBase()
+    {
+        StopRmt();
+        free(_pixels);
+        free(_rmtBuffer);
+    }
+
+    bool IsReadyToUpdate()
+    {
+        // TODO: use "rmt_tx_end_callback" to set channel BUSY->IDLE state (?)
+        updateState();
+        return (_rmtState == NeoRmtState_Idle);
+    }
+
+    void Initialize()
+    {
+        ESP_ERROR_CHECK(get_channel_free(&_channel));
+
+        Serial.printf("[NeoPixelBus] Initializing RMT on channel:%u, pin:%u\n", _channel, _pin);
```

Comment:
don't leave serial debug outputs enabled by default.  Its ok to just comment them out though.